src/navi/core.clj

```diff
@@ -64,36 +57,35 @@
               .getSchema
               spec))))
 
-(defmulti spec class)
+(defmulti spec
+  (fn [^Schema schema]
+    (.getTypes schema)))
 
 (defmethod spec
-  StringSchema
-  [_]
-  string?)
+  #{"string"}
```

Comment:
what's the rationale for matching on a set instead of actual values like `"string"` or `"integer"`?

Reply:
Schema objects can have [multiple types](https://json-schema.org/understanding-json-schema/reference/type), so this was the most direct way to adapt the existing code without making too many changes. It would probably work fine to detect the case of multiple types and join the specs with a malli `:or` form. Then the multimethod could dispatch on the individual scalar values. Would you prefer that approach?